airflow/providers/databricks/sensors/databricks_partition.py

```diff
@@ -0,0 +1,157 @@
+        self.table_name = table_name
+        self.partition_name = partition_name
+        self.caller = caller
+        self.client_parameters = client_parameters or {}
+        self.hook_params = kwargs.pop("hook_params", {})
+        self.handler = handler
+        self.partition_operator = partition_operator
+
+    def _get_hook(self) -> DatabricksSqlHook:
+        return DatabricksSqlHook(
+            self.databricks_conn_id,
+            self._http_path,
+            self._sql_endpoint_name,
+            self.session_config,
+            self.http_headers,
+            self.catalog,
+            self.schema,
+            self.caller,
+            **self.client_parameters,
+            **self.hook_params,
+        )
+
+    def _sql_sensor(self, sql):
+        hook = self._get_hook()
+        sql_result = hook.run(
+            sql,
+            handler=self.handler if self.do_xcom_push else None,
+        )
+        return sql_result
+
+    def _check_table_partitions(self) -> list:
+        if self.catalog is not None:
+            complete_table_name = str(self.catalog + "." + self.schema + "." + self.table_name)
+            self.log.debug("Table name generated from arguments: %s", complete_table_name)
+        else:
+            raise AirflowException("Catalog name not specified, aborting query execution.")
+        partition_columns = self._sql_sensor(f"describe detail {complete_table_name}")[7]
+        self.log.debug("table_info: %s", partition_columns)
+        if len(partition_columns) < 1:
+            raise AirflowException("Table %s does not have partitions", complete_table_name)
+        partitions_list = []
+        for partition_col, partition_value in self.partition_name.items():
+            if partition_col in partition_columns:
+                if isinstance(partition_value, list):
+                    partitions_list.append(f"""{partition_col} in {tuple(partition_value)}""")
+                if isinstance(partition_value, (int, float, complex)):
+                    partitions_list.append(f"""{partition_col}{self.partition_operator}{partition_value}""")
+                if isinstance(partition_value, str):
+                    partitions_list.append(
+                        f"""{partition_col}{self.partition_operator}\"{partition_value}\""""
+                    )
+            else:
+                raise AirflowException(
+                    "Column %s not part of table partitions: %s", partition_col, partition_columns
+                )
+        partitions = " AND ".join(partitions_list)
+        partition_sql = f"SELECT 1 FROM {complete_table_name} WHERE {partitions}"
+        return self._sql_sensor(partition_sql)
+
+    def poke(self, context: Context) -> bool:
+        result = self._check_table_partitions()
+        self.log.debug("Partition sensor result: %s", result)
+        if len(result) < 1:
+            raise AirflowException("Databricks SQL partition sensor failed.")
+        return True
```

Comment:
We can use the same function defined in the parent, but simply have an overridable function something like, `_get_results` that will implement the logic.